src/IR.h

```diff
@@ -886,16 +886,44 @@ struct Atomic : public StmtNode<Atomic> {
     static const IRNodeType _node_type = IRNodeType::Atomic;
 };
 
-/** Represent a target-specific vector instruction.
- *  Intrinsic may not be element-wise operation, i.e.
- *  dot_products. */
-struct VectorIntrinsic : public ExprNode<VectorIntrinsic> {
-    std::string name;
+/** Represent a length-agnostic and target-specific
```

Comment:
Another question is whether it is worth adding information, possibly optional, that describes the shape change for each instruction as a function of its argument shapes. Some are standard coherent shapes, some are pure reductions, etc.

Reply:
That could be interesting, though I'm not sure I know of a use case where that information is needed. Do you have an idea of some sort of analysis where this information is useful?

For x86, I would probably have `elementwise`, `reduce2`, `reduce4`, `accumulate_reduce2`, and `accumulate_reduce4`. I suspect that this list would grow quite quickly, so I am hesitant to attempt to track this information without concrete use cases